raft/src/main/java/org/apache/kafka/snapshot/RecordsSnapshotReader.java

```diff
@@ -62,12 +62,7 @@ public int lastContainedLogEpoch() {
     public long lastContainedLogTimestamp() {
         if (!lastContainedLogTimestamp.isPresent()) {
             nextBatch.ifPresent(batch -> {
-                throw new IllegalStateException(
-                    String.format(
-                        "nextBatch was present when last contained log timestamp was not present",
-                        batch
-                    )
-                );
+                throw new IllegalStateException("nextBatch was present when last contained log timestamp was not present");
```

Comment:
This is the change that inspired the whole PR.
Currently, `String.format` is called but there is nothing to format and `batch` is simply ignored. We could possibly print the batch in question, but I guess it would be too verbose. Hence I decided to change the code to reflect the current behaviour (no printing of the batch in question).

Reply:
Nice catch! But I think the `batch` content is good for troubleshooting.

Reply:
Shall we print a `.toString` of the `batch`?

Reply:
No need.

Reply:
Added a method in `Batch` to describe it (without the records) and added this info in the exception. Let me know if this works